torch/library.py

```diff
@@ -93,8 +96,9 @@ def impl(self, op_name, fn, dispatch_key=''):
             >>>     return self * (1 / other)
             >>> my_lib.impl("div.Tensor", div_cpu, "CPU")
         '''
-        if not callable(fn):
-            raise TypeError(f"Input function is required to be a callable but found type {type(fn)}")
+        if not (callable(fn) or isinstance(fn, Fallthrough)):
+            raise TypeError("Input function is required to be a callable or a "
+                            f"torch.library.Fallthrough object but found type {type(fn)}")
```

Comment:
I know I suggested the Fallthrough object, but just to do a quick bit of bikeshedding:

There are a couple of designs we could do:
1. The Fallthrough object
2. User passes "fallthrough" instead of a function
3. Add a Library.fallthrough(dispatchkey) API.

I like (1) or (2) because they match the C++ TORCH_LIBRARY API. Thoughts?

Reply:
Why not `torch.library.fallthrough_kernel` where we can check `if fn is torch.library.fallthrough_kernel`?

Reply:
4. A fallthrough function instead of a fallthrough class